File: src/main/java/com/tobe/healthy/notification/domain/dto/in/CommandSendNotification.java
package com.tobe.healthy.notification.domain.dto.in;

import com.tobe.healthy.notification.domain.entity.NotificationCategory;
import com.tobe.healthy.notification.domain.entity.NotificationType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CommandSendNotification {

    private String title;
    private String content;
    private List<Long> receiverIds;
    private NotificationType notificationType;
    private NotificationCategory notificationCategory;
    private Long targetId;
    private String clickUrl;
    private Long studentId;
    private String studentName;
}

File: src/main/kotlin/com/tobe/healthy/notification/domain/dto/in/CommandSendNotification.kt
package com.tobe.healthy.notification.domain.dto.`in`

import com.tobe.healthy.notification.domain.entity.NotificationCategory
import com.tobe.healthy.notification.domain.entity.NotificationType

data class CommandSendNotification(
    val title: String,
    val content: String,
    val receiverIds: List<Long>,
    val notificationType: NotificationType,
    val notificationCategory: NotificationCategory,
    val targetId: Long? = null,
    val clickUrl: String? = null,
    val studentId: Long? = null,
    val studentName: String? = null
)